develop/CICD/Tutorials/CICD_For_Other_Items/CICD_Tutorial_For_Other_Items_Multi-Artifact_DataMiner_Package_VisualStudio_And_GitHub.md

```diff
@@ -0,0 +1,212 @@
+         }
+         catch (Exception e)
+         {
+            engine.ExitFail($"Exception encountered during installation: {e}");
+         }
+      ```
+
+   1. Save the changes
+
+## Step 3: Create a GitHub repository
+
+1. In the menu bar of Visual Studio, select *GIT* > *Create GIT Repository*.
+
+1. Create a new GitHub repository:
+
+   1. Select a name, specify your GitHub account, and mark yourself as the owner.
+
+   1. When you have entered all the information, click *Create and Push*.
+
+1. Go to the newly created GitHub repository on <https://github.com/>.
+
+   > [!TIP]
+   > In Visual Studio 2022, you can open the *GIT* menu and select *GitHub/View Pull Requests* to quickly access the correct repository.
+
+## Step 4: Check the GitHub Actions
+
+1. In GitHub, go to the *Actions* tab.
+
+1. Click the workflow run that failed (usually called *Add project files*).
+
+1. Click the "build" step that failed and read the failing error.
+
+   ``` text
+   Error: DATAMINER_TOKEN is not set. Release not possible!
+   Please create or re-use an admin.dataminer.services token by visiting: https://admin.dataminer.services/.
+   Navigate to the right Organization then go to Keys and create/find a key with permissions to Register Catalog Items.
+   Copy the value of the token.
+   Then set a DATAMINER_TOKEN secret in your repository settings: <Dynamic Link>
+   ```
+
+   You can use the links from the actual error in the next couple of steps.
+
+## Step 5: Create and add a secret
+
+1. Obtain an **organization key** from [admin.dataminer.services](https://admin.dataminer.services/) with the required scopes:
+
+   - *Register catalog items*
+   - *Read catalog items*
+
+   > [!TIP]
+   > See also: [dataminer.services keys](xref:GitHub_Secrets#dataminerservices-keys)
+
+1. Add the key as a secret in your GitHub repository, by navigating to *Settings* > *Secrets and variables* > *Actions* and creating a secret named `DATAMINER_TOKEN`.
+
+1. Re-run the workflow.
+
+With this setup, any push with new content (including the initial creation) to the main/master branch will generate a new pre-release version, using the latest commit message as the version description.
+
+## Step 6: Check the results
+
+1. Go to the [DataMiner Catalog](https://catalog.dataminer.services/)
+
+1. Check in the top-right corner if the correct organization is selected.
+
+1. Search for the name of your package.
+
+   By default, this is your Visual Studio project name (e.g. *TutorialPackage*).
+
+1. Go to the *VERSIONS* tab.
+
+   This tab should contain the new version of the package, which is now available for deployment from the Catalog.
+
+1. If you have an available non-production DataMiner Agent, use the *Deploy* button to deploy the new version to the DMA.
+
+   All artifacts will be installed, and you should see an information event saying *Tutorial installation ran with config: Hello World!*.
```

Comment:
Is this an information event in the Cube Alarm Console?

Reply:
Yes. from the engine.generateInformation  call in the earlier code.